# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../gsmlib/gcc43.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -ruN gsmlib-1.10-orig/apps/gsmsmsd.cc gsmlib-1.10/apps/gsmsmsd.cc
--- gsmlib-1.10-orig/apps/gsmsmsd.cc	2002-10-03 11:55:15.000000000 +0200
+++ gsmlib-1.10/apps/gsmsmsd.cc	2010-08-25 12:36:42.500283559 +0200
@@ -14,6 +14,7 @@
 #include <gsm_config.h>
 #endif
 #include <gsmlib/gsm_nls.h>
+#include <string.h>
 #include <string>
 
 #ifdef WIN32
diff -ruN gsmlib-1.10-orig/gsmlib/gsm_me_ta.cc gsmlib-1.10/gsmlib/gsm_me_ta.cc
--- gsmlib-1.10-orig/gsmlib/gsm_me_ta.cc	2002-10-22 22:48:39.000000000 +0200
+++ gsmlib-1.10/gsmlib/gsm_me_ta.cc	2010-08-25 12:36:42.504279967 +0200
@@ -19,6 +19,8 @@
 #include <gsmlib/gsm_parser.h>
 #include <gsmlib/gsm_sysdep.h>
 
+#include <stdlib.h>
+
 using namespace std;
 using namespace gsmlib;
 
diff -ruN gsmlib-1.10-orig/gsmlib/gsm_sms_codec.cc gsmlib-1.10/gsmlib/gsm_sms_codec.cc
--- gsmlib-1.10-orig/gsmlib/gsm_sms_codec.cc	2002-09-23 00:09:16.000000000 +0200
+++ gsmlib-1.10/gsmlib/gsm_sms_codec.cc	2010-08-25 12:36:42.500283559 +0200
@@ -18,6 +18,7 @@
 #include <gsmlib/gsm_sms_codec.h>
 #include <gsmlib/gsm_util.h>
 #include <time.h>
+#include <limits.h>
 #include <strstream>
 #include <iomanip>
 #ifdef HAVE_STRING_H
diff -ruN gsmlib-1.10-orig/gsmlib/gsm_sorted_phonebook.cc gsmlib-1.10/gsmlib/gsm_sorted_phonebook.cc
--- gsmlib-1.10-orig/gsmlib/gsm_sorted_phonebook.cc	2002-05-14 21:38:12.000000000 +0200
+++ gsmlib-1.10/gsmlib/gsm_sorted_phonebook.cc	2010-08-25 12:36:42.504279967 +0200
@@ -20,6 +20,7 @@
 #include <iostream>
 #include <fstream>
 #include <limits.h>
+#include <string.h>
 
 const int MAX_LINE_SIZE = 1000;
 
diff -ruN gsmlib-1.10-orig/gsmlib/gsm_unix_serial.cc gsmlib-1.10/gsmlib/gsm_unix_serial.cc
--- gsmlib-1.10-orig/gsmlib/gsm_unix_serial.cc	2002-09-22 13:51:10.000000000 +0200
+++ gsmlib-1.10/gsmlib/gsm_unix_serial.cc	2010-08-25 12:36:42.500283559 +0200
@@ -21,6 +21,7 @@
 #include <iostream>
 #include <strstream>
 #include <errno.h>
+#include <string.h>
 #include <stdio.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
diff -ruN gsmlib-1.10-orig/gsmlib/gsm_util.cc gsmlib-1.10/gsmlib/gsm_util.cc
--- gsmlib-1.10-orig/gsmlib/gsm_util.cc	2002-09-22 15:41:08.000000000 +0200
+++ gsmlib-1.10/gsmlib/gsm_util.cc	2010-08-25 12:36:42.500283559 +0200
@@ -17,6 +17,7 @@
 #include <gsmlib/gsm_util.h>
 #include <gsmlib/gsm_sysdep.h>
 #include <sys/stat.h>
+#include <stdlib.h>
 #include <assert.h>
 #include <string.h>
 #include <iostream>
